backend/service/k8s/helper.go

```diff
@@ -0,0 +1,17 @@
+package k8s
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+)
+
+// GenerateStrategicPatch will return a patch that yields
+// the modified object when applied to the original object,
+// or an error if either of the two objects is invalid.
+func GenerateStrategicPatch(oldBytes, newBytes []byte) ([]byte, error) {
+	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldBytes, newBytes, appsv1.Deployment{})
```

Comment:
this is hardcoded to appsv1.Deployment, will it work for other types?

i think a better strategy for this func is to use a signature like:

```go
func GenerateStrategicPatch(original, modified runtime.Object) ([]byte, error)
```

pushing the json marshaling etc down and reflecting on the type within the function

Reply:
updated this method to be `GenerateStrategicPatch(original, modified runtime.Object, dataStruct interface{})` to take in the object type as the third argument. This will avoid the need for us to have a long switch statement in the method like:
```
switch original.(type) {
        case *appsv1.Deployment:
            patchBytes, err = strategicpatch.CreateTwoWayMergePatch(oldBytes, newBytes, appsv1.Deployment{})
        case *appsv1.StatefulSet:
            patchBytes, err = strategicpatch.CreateTwoWayMergePatch(oldBytes, newBytes, appsv1.StatefulSet{})
        case *appsv1.DaemonSet:
            patchBytes, err = strategicpatch.CreateTwoWayMergePatch(oldBytes, newBytes, appsv1.DaemonSet{})
....
```

Reply:
@danielhochman let me know if you can suggest a cleaner way to initialize the object type thats passed in without the 3rd argument in the method

Reply:
Note: re-test manually with the new changes